2025-07-08 23:42:08,577 - asyncio - DEBUG - Using selector: KqueueSelector
2025-07-08 23:42:08,577 - root - DEBUG - Starting bot
2025-07-08 23:42:08,577 - root - DEBUG - conversation_choose_whale
2025-07-08 23:42:08,595 - aiogram.dispatcher - INFO - Start polling
2025-07-08 23:42:09,174 - aiogram.dispatcher - INFO - Run polling for bot @bImas_tas_bot id=7964566377 - 'bImas_tas_bot'
2025-07-08 23:42:09,310 - aiogram.event - INFO - Update id=996116841 is not handled. Duration 1 ms by bot id=7964566377
2025-07-08 23:42:09,310 - aiogram.event - INFO - Update id=996116842 is not handled. Duration 1 ms by bot id=7964566377
2025-07-08 23:42:45,156 - root - DEBUG - cmd_start from 382371597
2025-07-08 23:42:45,168 - root - INFO - Use hardcoded cashier for 382371597
2025-07-08 23:42:45,816 - aiogram.event - INFO - Update id=996116843 is handled. Duration 660 ms by bot id=7964566377
2025-07-08 23:42:48,366 - root - DEBUG - stoplist_start from 382371597
2025-07-08 23:42:48,367 - root - DEBUG - Postgre query: 
        SELECT nomenclature_name
        FROM stop_list
        INNER JOIN nomenclature USING(iiko_code)
        WHERE id_store = 0 AND date_end_stop is NULL;
    
2025-07-08 23:42:48,981 - root - DEBUG - Result: []
2025-07-08 23:42:49,870 - aiogram.event - INFO - Update id=996116844 is handled. Duration 1504 ms by bot id=7964566377
2025-07-08 23:42:52,497 - root - DEBUG - stoplist_add query=add
2025-07-08 23:42:52,497 - root - DEBUG - Postgre query: 
        (SELECT nomenclature_name
            FROM nomenclature
            WHERE stop_list is TRUE
            EXCEPT
            SELECT nomenclature_name
            FROM stop_list
            INNER JOIN store USING(id_store)
            INNER JOIN nomenclature USING(iiko_code)
            WHERE id_store = 0 AND date_end_stop is NULL)
            INTERSECT
            SELECT nomenclature_name
            FROM store_nomenclature
            INNER JOIN nomenclature USING(iiko_code)
            WHERE id_store = 0
            ORDER BY nomenclature_name;
    
2025-07-08 23:42:52,497 - root - INFO - TEST_MODE: skip query execution
2025-07-08 23:42:52,664 - root - DEBUG - _to_dispatch info=None
2025-07-08 23:42:52,664 - root - DEBUG - Postgre query: 
        SELECT nomenclature_name
        FROM stop_list
        INNER JOIN nomenclature USING(iiko_code)
        WHERE id_store = 0 AND date_end_stop is NULL;
    
2025-07-08 23:42:53,319 - root - DEBUG - Result: []
2025-07-08 23:42:53,525 - aiogram.event - INFO - Update id=996116845 is not handled. Duration 1029 ms by bot id=7964566377
2025-07-08 23:42:53,525 - aiogram.event - ERROR - Cause exception while process update id=996116845 by bot id=7964566377
TelegramBadRequest: Telegram server says - Bad Request: message is not modified: specified new message content and reply markup are exactly the same as a current content and reply markup of the message
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/dispatcher.py", line 309, in _process_update
    response = await self.feed_update(bot, update, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/dispatcher.py", line 158, in feed_update
    response = await self.update.wrap_outer_middleware(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/middlewares/error.py", line 25, in __call__
    return await handler(event, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/middlewares/user_context.py", line 49, in __call__
    return await handler(event, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/fsm/middleware.py", line 42, in __call__
    return await handler(event, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/event/telegram.py", line 121, in trigger
    return await wrapped_inner(event, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/event/handler.py", line 43, in call
    return await wrapped()
           ^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/dispatcher.py", line 276, in _listen_update
    return await self.propagate_event(update_type=update_type, event=event, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 142, in propagate_event
    return await observer.wrap_outer_middleware(_wrapped, event=event, data=kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 137, in _wrapped
    return await self._propagate_event(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 170, in _propagate_event
    response = await router.propagate_event(update_type=update_type, event=event, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 142, in propagate_event
    return await observer.wrap_outer_middleware(_wrapped, event=event, data=kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 137, in _wrapped
    return await self._propagate_event(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 162, in _propagate_event
    response = await observer.trigger(event, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/event/telegram.py", line 121, in trigger
    return await wrapped_inner(event, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/event/handler.py", line 43, in call
    return await wrapped()
           ^^^^^^^^^^^^^^^
  File "/Users/konstantindashkecich/Downloads/SuperBot2_2/Conversations/conversationStoplist.py", line 153, in stoplist_add
    await _to_dispatch(query.message, state)
  File "/Users/konstantindashkecich/Downloads/SuperBot2_2/Conversations/conversationStoplist.py", line 92, in _to_dispatch
    await message.edit_text(text=text, reply_markup=keyboard_stop_actions(bool(stop_list)))
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/methods/base.py", line 84, in emit
    return await bot(self)
           ^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/client/bot.py", line 508, in __call__
    return await self.session(self, method, timeout=request_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/client/session/base.py", line 254, in __call__
    return cast(TelegramType, await middleware(bot, method))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/client/session/aiohttp.py", line 178, in make_request
    response = self.check_response(
               ^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/client/session/base.py", line 120, in check_response
    raise TelegramBadRequest(method=method, message=description)
aiogram.exceptions.TelegramBadRequest: Telegram server says - Bad Request: message is not modified: specified new message content and reply markup are exactly the same as a current content and reply markup of the message
2025-07-08 23:43:47,300 - root - DEBUG - register_cancel
2025-07-08 23:43:47,301 - root - INFO - Use hardcoded cashier for 382371597
2025-07-08 23:43:47,963 - aiogram.event - INFO - Update id=996116846 is handled. Duration 664 ms by bot id=7964566377
2025-07-08 23:43:50,141 - root - DEBUG - stoplist_start from 382371597
2025-07-08 23:43:50,141 - root - INFO - Use hardcoded cashier for 382371597
2025-07-08 23:43:50,328 - aiogram.event - INFO - Update id=996116847 is handled. Duration 189 ms by bot id=7964566377
2025-07-08 23:43:54,949 - root - DEBUG - stoplist_start from 382371597
2025-07-08 23:43:54,949 - root - INFO - Use hardcoded cashier for 382371597
2025-07-08 23:43:55,240 - aiogram.event - INFO - Update id=996116848 is handled. Duration 294 ms by bot id=7964566377
2025-07-08 23:44:00,707 - aiogram.event - INFO - Update id=996116849 is not handled. Duration 2 ms by bot id=7964566377
2025-07-08 23:44:03,774 - root - DEBUG - stoplist_start from 382371597
2025-07-08 23:44:03,774 - root - INFO - Use hardcoded cashier for 382371597
2025-07-08 23:44:03,978 - aiogram.event - INFO - Update id=996116850 is handled. Duration 204 ms by bot id=7964566377
2025-07-08 23:44:29,471 - root - DEBUG - cmd_start from 382371597
2025-07-08 23:44:29,471 - root - INFO - Use hardcoded cashier for 382371597
2025-07-08 23:44:30,163 - aiogram.event - INFO - Update id=996116851 is handled. Duration 693 ms by bot id=7964566377
2025-07-08 23:44:31,311 - root - DEBUG - stoplist_start from 382371597
2025-07-08 23:44:31,311 - root - DEBUG - Postgre query: 
        SELECT nomenclature_name
        FROM stop_list
        INNER JOIN nomenclature USING(iiko_code)
        WHERE id_store = 0 AND date_end_stop is NULL;
    
2025-07-08 23:44:31,884 - root - DEBUG - Result: []
2025-07-08 23:44:32,521 - aiogram.event - INFO - Update id=996116852 is handled. Duration 1211 ms by bot id=7964566377
2025-07-08 23:44:33,972 - root - DEBUG - stoplist_add query=add
2025-07-08 23:44:33,972 - root - DEBUG - Postgre query: 
        (SELECT nomenclature_name
            FROM nomenclature
            WHERE stop_list is TRUE
            EXCEPT
            SELECT nomenclature_name
            FROM stop_list
            INNER JOIN store USING(id_store)
            INNER JOIN nomenclature USING(iiko_code)
            WHERE id_store = 0 AND date_end_stop is NULL)
            INTERSECT
            SELECT nomenclature_name
            FROM store_nomenclature
            INNER JOIN nomenclature USING(iiko_code)
            WHERE id_store = 0
            ORDER BY nomenclature_name;
    
2025-07-08 23:44:33,972 - root - INFO - TEST_MODE: skip query execution
2025-07-08 23:44:34,195 - root - DEBUG - _to_dispatch info=None
2025-07-08 23:44:34,195 - root - DEBUG - Postgre query: 
        SELECT nomenclature_name
        FROM stop_list
        INNER JOIN nomenclature USING(iiko_code)
        WHERE id_store = 0 AND date_end_stop is NULL;
    
2025-07-08 23:44:35,823 - root - DEBUG - Result: []
2025-07-08 23:44:36,102 - aiogram.event - INFO - Update id=996116853 is not handled. Duration 2131 ms by bot id=7964566377
2025-07-08 23:44:36,103 - aiogram.event - ERROR - Cause exception while process update id=996116853 by bot id=7964566377
TelegramBadRequest: Telegram server says - Bad Request: message is not modified: specified new message content and reply markup are exactly the same as a current content and reply markup of the message
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/dispatcher.py", line 309, in _process_update
    response = await self.feed_update(bot, update, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/dispatcher.py", line 158, in feed_update
    response = await self.update.wrap_outer_middleware(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/middlewares/error.py", line 25, in __call__
    return await handler(event, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/middlewares/user_context.py", line 49, in __call__
    return await handler(event, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/fsm/middleware.py", line 42, in __call__
    return await handler(event, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/event/telegram.py", line 121, in trigger
    return await wrapped_inner(event, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/event/handler.py", line 43, in call
    return await wrapped()
           ^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/dispatcher.py", line 276, in _listen_update
    return await self.propagate_event(update_type=update_type, event=event, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 142, in propagate_event
    return await observer.wrap_outer_middleware(_wrapped, event=event, data=kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 137, in _wrapped
    return await self._propagate_event(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 170, in _propagate_event
    response = await router.propagate_event(update_type=update_type, event=event, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 142, in propagate_event
    return await observer.wrap_outer_middleware(_wrapped, event=event, data=kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 137, in _wrapped
    return await self._propagate_event(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/router.py", line 162, in _propagate_event
    response = await observer.trigger(event, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/event/telegram.py", line 121, in trigger
    return await wrapped_inner(event, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/dispatcher/event/handler.py", line 43, in call
    return await wrapped()
           ^^^^^^^^^^^^^^^
  File "/Users/konstantindashkecich/Downloads/SuperBot2_2/Conversations/conversationStoplist.py", line 153, in stoplist_add
    await _to_dispatch(query.message, state)
  File "/Users/konstantindashkecich/Downloads/SuperBot2_2/Conversations/conversationStoplist.py", line 92, in _to_dispatch
    await message.edit_text(text=text, reply_markup=keyboard_stop_actions(bool(stop_list)))
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/methods/base.py", line 84, in emit
    return await bot(self)
           ^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/client/bot.py", line 508, in __call__
    return await self.session(self, method, timeout=request_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/client/session/base.py", line 254, in __call__
    return cast(TelegramType, await middleware(bot, method))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/client/session/aiohttp.py", line 178, in make_request
    response = self.check_response(
               ^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/aiogram/client/session/base.py", line 120, in check_response
    raise TelegramBadRequest(method=method, message=description)
aiogram.exceptions.TelegramBadRequest: Telegram server says - Bad Request: message is not modified: specified new message content and reply markup are exactly the same as a current content and reply markup of the message
